bitrepository-client/src/main/java/org/bitrepository/client/conversation/mediator/CollectionBasedConversationMediator.java

```diff
@@ -149,15 +154,20 @@ private final class ConversationCleaner extends TimerTask {
         @Override
         public void run() {
             Conversation[] conversationArray = conversations.values().toArray(new Conversation[0]);
-            long currentTime = System.currentTimeMillis();
+            Duration conversationTimeout = XmlUtils.xmlDurationToDuration(
+                    settings.getReferenceSettings().getClientSettings().getConversationTimeout());
+            Instant currentTime = Instant.now();
             for (Conversation conversation : conversationArray) {
                 if (conversation.hasEnded()) {
                     conversations.remove(conversation.getConversationID());
-                } else if (currentTime - conversation.getStartTime() >
-                        settings.getReferenceSettings().getClientSettings().getConversationTimeout().longValue()) {
-                    log.warn("Failing timed out conversation " + conversation.getConversationID() + " " + "(Age " +
-                            (currentTime - conversation.getStartTime()) + "ms)");
-                    failConversation(conversation, "Failing timed out conversation " + conversation.getConversationID());
+                } else {
+                    Instant startTime = Instant.ofEpochMilli(conversation.getStartTime());
```

Comment:
Eventually I think it would be nice if `long`-variables indicating some form of time/duration like `Conversation#getStartTime()` would also be updated to use java.time so intermediate conversions like this wouldn't be necessary.

For now I think it's fine though.

Reply:
I certainly agree: ideally we don’t want to use `long` neither for a point in time nor for a duration/interval (except when calling a library method that requires it, then we need to convert). I drew a line somewhere for now, and the place to draw it always a matter of weighing. Thanks for the thought.